### DISCUSSION TITLE
Fetch entity when not found with a local resolver

### DISCUSSION THREAD

**Discussion by oreqizer** [urql-graphql/urql#1762] [Q&A] [answered]
Is there a way to make URLQ fetch an entity if it is not found by a local resolver?

I have a resolver that looks like the following:

```ts
  resolvers: {
    Query: {
      search: relayPagination(),
      program: (parent, vars, c, info) => {
        // I want to display a partial result from the "search" query when transitioning to the "program" page
        // that searches results by "slug", not "id" (in the URL)
        const found = c
          .resolve("Query", "search")
          ?.find(key => c.resolve(key, "slug") === vars.slug);

        if (found != null) {
          return found;
        }
        return null; // ???
      },
    },
```

thing is, when I go to the "program" page from "search", it works, because `found` is not  `null` in the resolver. But when I visit the "program" page directly, `null` is returned, and the result is never fetched.

How do I force an entity fetch, when `found` is `null`?

**EDIT**

Returning `undefined` acts the same way

**Comment by kitten** [comment, marked as answer]
It's important to note that `undefined` and `null` are handled differently in Graphcache (and GraphQL in general).
Because GraphQL is serialised as JSON, `null` is on every field a valid property value that signifies that `null` was explicitly returned by the API. The exception is that a schema may dictate which fields are non-nullable. In schema awareness mode this is turned a bit on its head in Graphcache, but it checks which fields can remain `null`.

Either way, if you return `null` in your resolver then Graphcache will assume that the value of this field is explicitly `null`. However if you return `undefined` in your resolver then you're signalling that there is _no cached value_ for this field, which will trigger a reexecution of the operation against the API, i.e. typically a fetch.

**Reply by oreqizer** [reply]
ah ok, will try! I didn't do `undefined`, because types tell me I can't 

<img width="798" alt="Screenshot 2021-07-02 at 14 41 21" src="https://user-images.githubusercontent.com/11172408/124276213-cdfaf580-db43-11eb-906a-b7f7845b2103.png">

**Reply by oreqizer** [reply]
just tried it, and even then, returning `undefined` did nothing 😕 acts the same as returning `null` — the query just never fetches
